From 98c59526ff12507e9a984dfca1be88110f4d0124 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tomi=20Lepp=C3=A4nen?= <tomi@tomin.site>
Date: Fri, 2 Apr 2021 22:10:08 +0300
Subject: [PATCH] Use let* instead of letrec

letrec seems to cause an exception with this version of guile.
---
 games/hamilton.scm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/aisleriot/games/hamilton.scm b/aisleriot/games/hamilton.scm
index 6ee557c3..ae4dace8 100644
--- a/aisleriot/games/hamilton.scm
+++ b/aisleriot/games/hamilton.scm
@@ -290,7 +290,7 @@
 ; start to end. If the end slot is a foundation, the hint is just the first
 ; card. If in the tableau, the hint must be the longest moveable list of cards.
 (define (maybe-hint p)
-  (letrec ((start (car p))
+  (let* ((start (car p))
            (end (cdr p))
            (cards (if (member end foundation)
 		      (list (get-top-card start))
-- 
2.31.1

